### ISSUE TITLE
(Most of) our C++ processes don't self daemonize

### ISSUE THREAD

**Issue by bossmc** [Metaswitch/cpp-common#473]
Only Sprout/Bono call into https://github.com/Metaswitch/sprout/blob/dev/src/main.cpp#L1108 to background themselves at boot.  Other C++ processes use `start-stop-daemon --background` to achieve this, but the man page for that option is pretty damning:

>       -b, --background
>              Typically  used  with  programs  that don't detach on their own.
>              This option will force start-stop-daemon to fork before starting
>              the  process,  and  force  it  into  the  background.   WARNING:
>              start-stop-daemon cannot check the exit status  if  the  process
>              fails  to  execute for any reason. This is a last resort, and is
>              only meant for programs that either make  no  sense  forking  on
>              their  own,  or where it's not feasible to add the code for them
>              to do this themselves.

We should move the forking logic to cpp-common and use it in all C++ services (as we already do for python ones).

**Comment by chris-elford-metaswitch** [comment]
Should we? My instinct here is to move everything to use start-stop-daemon and dispense with our own forking code. That may be naïve

**Comment by mirw** [comment]
Regarding moving to `start-stop-daemon`, the section Rob pulls out above says:

```
WARNING:
         start-stop-daemon cannot check the exit status  if  the  process
         fails  to  execute for any reason. This is a last resort, and is
         only meant for programs that either make  no  sense  forking  on
         their  own,  or where it's not feasible to add the code for them
         to do this themselves.
```

....so I think if we can do better (and we can), we should.

**Comment by chris-elford-metaswitch** [comment]
Fair enough

**Comment by bossmc** [comment]
The basic fix is in https://github.com/Metaswitch/cpp-common/pull/491 and https://github.com/Metaswitch/sprout/pull/1374, this needs to be ported to the other C++ binaries.

**Comment by bossmc** [comment]
Fixed by #491 and friends.
